### PR TITLE
SCT-156 Build: ErrorHandlerMiddleware pattern in Bean framework

### PR DESCRIPTION
**Pull request by gaoweiqiao-tre**
- releated task is [Build: ErrorHandlerMiddleware pattern in Bean framework](https://retailai.atlassian.net/browse/SCT-156)
- develop error handler middleware
- unit test is in `error_middleware_test.go`
image

### CODE REVIEW

File: internal/project/framework/bean.go [hunk 1]
```diff
@@ -44,3 +53,20 @@ func (b *Bean) Bootstrap() {
...
+	b.errorHandlerMiddlewares = append(b.errorHandlerMiddlewares, errorHandlerMiddleware)
+}
+
+func concatErrorHandlerMiddlewares(middlewares []berror.ErrorHandlerMiddleware, frameworkMiddlewares ...berror.ErrorHandlerMiddleware) []berror.ErrorHandlerMiddleware {
+	if middlewares == nil {
+		middlewares = []berror.ErrorHandlerMiddleware{}
+	}
+	for _, m := range frameworkMiddlewares {
+		middlewares = append(middlewares, m)
+	}
```
Comment (yu-yk) [inline internal/project/framework/bean.go:70]: `middlewares = append(middlewares, frameworkMiddlewares...)`

File: internal/project/framework/bean.go [hunk 2]
```diff
@@ -10,21 +10,30 @@ import (
...
 
 func (b *Bean) Bootstrap() {
 	// Create a new echo instance
 	e := bootstrap.New()
 
+	errorHandlerMiddlewares := concatErrorHandlerMiddlewares(b.errorHandlerMiddlewares,
+		berror.ValidationErrorHanderMiddleware,
+		berror.APIErrorHanderMiddleware,
+		berror.HTTPErrorHanderMiddleware,
+		berror.DefaultErrorHanderMiddleware)
```
Comment (yu-yk) [inline internal/project/framework/bean.go:35]: we can also use
```
b.UseErrorHandlerMiddleware(berror.ValidationErrorHanderMiddleware)
b.UseErrorHandlerMiddleware(berror.APIErrorHanderMiddleware)
...
```
then we don't need the concat function, right?

**Review by yu-yk — CHANGES_REQUESTED** [review]

File: internal/project/framework/bean.go (hunk 1 above)
Comment (yu-yk) [inline internal/project/framework/bean.go:70]:
```suggestion
	middlewares = append(middlewares, frameworkMiddlewares...)
```

File: internal/project/framework/bean.go (hunk 2 above)
Comment (gaoweiqiao-tre) [reply internal/project/framework/bean.go:35]: yes, but it make user don't need to think about framework errorHandlers, and  it make less code in `main.go`

Comment (gaoweiqiao-tre) [reply internal/project/framework/bean.go:35]: > we can also use
> 
> ```
> b.UseErrorHandlerMiddleware(berror.ValidationErrorHanderMiddleware)
> b.UseErrorHandlerMiddleware(berror.APIErrorHanderMiddleware)
> ...
> ```
> 
> then we don't need the concat function, right?

yes, but it make user don't need to think about framework errorHandlers, and  it make less code in `main.go`

**Review by yu-yk — APPROVED** [review]

**Pull request merged**